gh-actions/common/build-debian/action.yml

```diff
@@ -46,69 +55,194 @@ runs:
       run: |
         set -eu
 
-        echo "::group::Install devscripts"
-        DEBIAN_FRONTEND=noninteractive sudo apt update
-        DEBIAN_FRONTEND=noninteractive sudo apt install -y devscripts
-        echo "::endgroup::"
-
-        echo "::group::Append commit SHA to local version"
+        echo "::group::Create local version with commit and docker container"
         cd '${{ inputs.source-dir }}'
-        sanitized_docker=$( echo "${{ inputs.docker-image }}" | sed 's/://' )
-        debchange --local "+${sanitized_docker}+${{ github.sha }}" "Github build. Job id: ${{ github.run_id }}. Attempt: ${{ github.run_number }}."
+
+        # Short commit to avoid "package-has-long-file-name"
+        echo VERSION_REF=$(echo ${{ github.sha }} | cut -c1-8)  >> $GITHUB_ENV
+
+        echo DEBFULLNAME="GitHub actions runner" >> $GITHUB_ENV
+        echo DEBEMAIL="noreply@github.com" >> $GITHUB_ENV
+
+        if git status --porcelain &>/dev/null; then
+          echo VERSION_REF=$(git rev-list --count HEAD)+$(git rev-parse --short HEAD) >> $GITHUB_ENV
+          echo DEBFULLNAME="$(git log -1 --format='%an' HEAD) - GH Action" >> $GITHUB_ENV
+          echo DEBEMAIL="$(git log -1 --format='%ae' HEAD)" >> $GITHUB_ENV
+        fi
         echo "::endgroup::"
 
+    - name: Prepare source package
+      uses: kohlerdominik/docker-run-action@v1.1.0
+      with:
+        image: ${{ inputs.docker-image }}
+        environment: |
+          TERM=dumb
```

Comment:
Well dpkg build may warn about `TERM` not being set, and having this in theory should make tools aware of the environment we are in.

It's not strictly needed but a nice to have one since we're so low level, but let me know if you prefer avoid it.

Reply:
check if there is a warning. If there is a warning, let’s set it, but here, on GitHub action, I haven’t seen it IIRC, so if no warning, let’s not predict the future :)

Reply:
eh, I'm sure I saw it while building... IIRC was on go packages, but I need to check this again as I don't see it in the test job.